src/EntityFramework/Core/Common/DbXmlEnabledProviderManifest.cs

```diff
@@ -77,6 +77,7 @@ public override string NamespaceName
         /// <param name="edmType">
         /// An <see cref="T:System.Data.Entity.Core.Metadata.Edm.EdmType" /> for which the facet descriptions are to be retrieved.
         /// </param>
+        [SuppressMessage("Microsoft.Performance", "CA1800:DoNotCastUnnecessarily", Justification = "Only casts twice in debug mode.")]
```

Comment:
I don't feel super strongly but I think in this case removing the assert would be better.